pkg/parser/instance/proto.go

```diff
@@ -0,0 +1,141 @@
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				value, err := node.GetExpandedStringValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfString(value))
+				return nil
+			})
+		case protoreflect.Int64Kind, protoreflect.Sint64Kind, protoreflect.Uint64Kind,
+			protoreflect.Fixed64Kind, protoreflect.Sfixed64Kind:
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				value, err := node.GetExpandedStringValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				parsedValue, err := strconv.ParseInt(value, 10, 64)
+				if strings.EqualFold(field.JSONName(), "memory") {
+					parsedValue, err = ParseMegaBytes(value)
+				}
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfInt64(parsedValue))
+				return nil
+			})
+		case protoreflect.Int32Kind, protoreflect.Sint32Kind, protoreflect.Uint32Kind,
+			protoreflect.Fixed32Kind, protoreflect.Sfixed32Kind:
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				value, err := node.GetExpandedStringValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				parsedValue, err := strconv.ParseInt(value, 10, 32)
+				if strings.EqualFold(field.JSONName(), "memory") {
+					parsedValue, err = ParseMegaBytes(value)
+				}
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfInt32(int32(parsedValue)))
+				return nil
+			})
+		case protoreflect.BoolKind:
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				evaluation, err := node.GetBoolValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfBool(evaluation))
+				return nil
+			})
+		case protoreflect.FloatKind:
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				value, err := node.GetExpandedStringValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				parsedValue, err := strconv.ParseFloat(value, 32)
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfFloat32(float32(parsedValue)))
+				return nil
+			})
+		case protoreflect.DoubleKind:
+			instance.OptionalField(nameable.NewSimpleNameable(field.JSONName()), schema.TodoSchema, func(node *node.Node) error {
+				value, err := node.GetExpandedStringValue(mergedEnv)
+				if err != nil {
+					return err
+				}
+				parsedValue, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					return err
+				}
+				instance.proto.Set(field, protoreflect.ValueOfFloat64(parsedValue))
+				return nil
+			})
+		case protoreflect.GroupKind, protoreflect.BytesKind:
+			// not supported
```

Comment:
Perhaps we should just `case default` here instead of specific types for clarity?

Reply:
IMO in this case it's better to explicitly declare two unsupported cases. If in future there will some new kinds then linter will make sure they'll be covered. 🤷‍♂️